presto-spi/src/main/java/com/facebook/presto/spi/type/FloatType.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.type;
+
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+
+import static com.facebook.presto.spi.StandardErrorCode.NUMERIC_VALUE_OUT_OF_RANGE;
+import static com.facebook.presto.spi.type.TypeSignature.parseTypeSignature;
+import static io.airlift.slice.SizeOf.SIZE_OF_FLOAT;
+
+public final class FloatType
+        extends AbstractFixedWidthType
+{
+    public static final FloatType FLOAT = new FloatType();
+
+    private FloatType()
+    {
+        super(parseTypeSignature(StandardTypes.FLOAT), long.class, SIZE_OF_FLOAT);
+    }
+
+    @Override
+    public long getLong(Block block, int position)
+    {
+        return (long) block.getInt(position, 0);
+    }
+
+    public void writeFloat(BlockBuilder blockBuilder, float value)
+    {
+        writeLong(blockBuilder, Float.floatToRawIntBits(value));
+    }
+
+    @Override
+    public void writeLong(BlockBuilder blockBuilder, long value)
+    {
+        try {
+            Math.toIntExact(value);
+        }
+        catch (ArithmeticException e) {
+            throw new PrestoException(NUMERIC_VALUE_OUT_OF_RANGE, "More than 32b value provided for Float type");
```

Comment:
Changed in 19ccfb2fed4ae97f9e978ca32890e2951e5ddc80.
Just to clarify, I'm not going to put any range values for float here (like it's for `Integer` (INT_MAX)) because user should never be able to provide a value which will produce more than 32b here:
1. on literal input, string is parsed by `Float.parseFloat` which will produce a valid float even for values not-accurately representable by IEE754 single-precision float number (will give some approximation in 32b float).
2. in terms of connectors, they should feed `FloatType` with 32b value, if not - data is corrupted / there's a software bug.

But I believe it's worth to have a sanity check here anyway, so I'm printing bit representation (instead of trying to interpret what is the 64b stream I got) to be able to debug the root cause of the problem (read above).